### PR TITLE
casmuser-2677: Add bonded NMN support, update bond pair selection

### PR DESCRIPTION
**Pull request by keopp-hpe**
## Summary and Scope

This PR adds bonded NMN support on UANs.  It also changes the way the bonded interfaces are selected for CAN.  Rather than try to divine which interfaces to use, the Admin must define them in Ansible.  This should be consistent for nodes that have equivalent hardware configurations.  Ansible host_vars may be used for deviations from the group_vars.

## Issues and Related PRs

* Resolves [CASMUSER-2677](https://jira-pro.its.hpecorp.net:8443/browse/CASMUSER-2677)
* Resolves [CASMUSER-2900](https://jira-pro.its.hpecorp.net:8443/browse/CASMUSER-2900)

### Tested on:

  * surtur

### Test description:

After the systems was configured to support NMN bonds on the switches, the updated uan_interfaces role was used to boot and configure the UAN with and without bonded NMN.

## Risks and Mitigations

Low risk

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [NA] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [NA] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: ansible/roles/uan_interfaces/tasks/main.yml [hunk 1]
```diff
@@ -276,4 +342,5 @@
       - "'Compute' not in group_names"
 
   when:
+    - not cray_cfs_image
```
Comment (alanm-hpe) [inline ansible/roles/uan_interfaces/tasks/main.yml:345]: This is controlled at the playbook level so I don't think we need this check.

File: ansible/roles/uan_interfaces/README.md [hunk 2]
```diff
@@ -15,6 +15,45 @@ Role Variables
...
+the node PXE boots.
+
+**IMPORTANT!** The first interface in the list must be the `nmn0` interface which is configured during the
+initial image boot, typically `ens10f0`.  This is required because the MAC address of the `nmn0` interface
+is the MAC associated with the IP address of the UAN.  The bonded `nmnb0` interface and the bond slaves
+will assume this MAC and the IP address of `nmn0` to preserve connectivity.
+
+The second interface is typically the first port of a different 25Gb NIC for resiliency.
+
+The default values of `uan_nmn_bond_slaves are shown here.  They may need to be changed to match the actual
```
Comment (alanm-hpe) [inline ansible/roles/uan_interfaces/README.md:48]: Missing a `

File: ansible/roles/uan_interfaces/README.md [hunk 3]
```diff
@@ -31,6 +70,28 @@ The default value of `uan_can_setup` is `no`.
...
+
+The interface names must be in a format which doesn't change between reboots of the node, such as `ens10f1`
+which is the second port of the NIC in slot 10.  
+
+**NOTE:** `ens10f1` is typically the second port of the OCP 25Gb card and is used as one of the bond
+slaves in the CAN `bond0` interface.
+
+The second interface is typically the second port of a different 25Gb NIC for resiliency.
+
+The default values of `uan_can_bond_slaves are shown here.  They may need to be changed to match the actual
```
Comment (alanm-hpe) [inline ansible/roles/uan_interfaces/README.md:86]: Missing closing `

File: ansible/roles/uan_interfaces/tasks/main.yml (hunk 1 above)
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/tasks/main.yml]: Fixed

File: ansible/roles/uan_interfaces/README.md (hunk 2 above)
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/README.md]: Fixed.

File: ansible/roles/uan_interfaces/README.md (hunk 3 above)
Comment (keopp-hpe) [reply ansible/roles/uan_interfaces/README.md]: Fixed.

**Review by alanm-hpe — APPROVED** [review]

**Pull request merged**